src/App.tsx

```diff
@@ -1,24 +1,88 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, {
+  useState,
+  useMemo,
+  useEffect,
+  useContext,
+} from 'react';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { Notification } from './components/Notification/Notification';
+import { getFilteredTodos } from './utils/getFilteredTodos';
+import { TodoContext } from './TodoContext';
+import * as todoService from './api/todos';
+import { TodoStatus } from './types';
+import {
+  USER_ID,
+  DOWNLOAD_ERROR,
+  POST_ERROR,
+} from './utils/constants';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [filterByStatus, setFilterByStatus] = useState(TodoStatus.All);
+
+  const {
+    todoItems,
+    setTodoItems,
+    setTempTodo,
+    setErrorMessage,
+  } = useContext(TodoContext);
```

Comment:
You can create a hook for context and use it, in this case, you don't need to pass the context every time. Example:
```javascript
const useTodo = () => useContext(TodoContext);
```